test/crdsvalidation/konnect.konghq.com/konnectcloudgatewaytransitgateway_test.go

```diff
@@ -0,0 +1,234 @@
+package crdsvalidation
+
+import (
+	"testing"
+
+	"github.com/samber/lo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	commonv1alpha1 "github.com/kong/kubernetes-configuration/api/common/v1alpha1"
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+	"github.com/kong/kubernetes-configuration/test/crdsvalidation/common"
+)
+
+var konnectTransitGatewayTypeMeta = metav1.TypeMeta{
+	APIVersion: konnectv1alpha1.GroupVersion.String(),
+	Kind:       "KonnectCloudGatewayTransitGateway",
+}
+
+func TestKonnectCloudGatewayTransitGateway(t *testing.T) {
+
+	var testAWSTransitGatewayConfig = &konnectv1alpha1.AWSTransitGateway{
+		Name: "aws-transit-gateway",
+		CIDRBlocks: []string{
+			"10.11.12.0/24",
+		},
+		AttachmentConfig: konnectv1alpha1.AwsTransitGatewayAttachmentConfig{
+			TransitGatewayID: "transit-gateway",
+			RAMShareArn:      "ram_share_arn",
+		},
+	}
+
+	var testAzureTransitGatewayConfig = &konnectv1alpha1.AzureTransitGateway{
+		Name: "azure-transit-gateway",
+		AttachmentConfig: konnectv1alpha1.AzureVNETPeeringAttachmentConfig{
+			TenantID:          "azure-tenant",
+			SubscriptionID:    "azure-subscription",
+			ResourceGroupName: "azure-resource-group",
+			VnetName:          "azure-vnet",
+		},
+	}
+
+	var namespacedNetworkRef = commonv1alpha1.ObjectRef{
+		Type: commonv1alpha1.ObjectRefTypeNamespacedRef,
+		NamespacedRef: &commonv1alpha1.NamespacedRef{
+			Name: "konnect-network",
+		},
+	}
+
+	t.Run("spec", func(t *testing.T) {
```

Comment:
Can we also add test cases for `KonnectCloudGatewayTransitGateway`s that do not cause errors? (I'd suggest 1 for each of the supported `spec.type`, WDYT?)

Reply:
Added the cases.